src/settings.c

```diff
@@ -516,6 +516,70 @@ void load_settings(char *cmdline_config_path)
                 );
 
         }
+
+        {
+                char *c = option_get_string(
+                        "mouse",
+                        "left_click", "-left_click", "close_current",
+                        "Action of Left click event"
+                );
+
+                if (strlen(c) > 0) {
+                        if (strcmp(c, "do_action") == 0)
```

Comment:
This is a lot of copy-paste code, it'll be easier to split this into a function (see the parse functions at the top of the file as an example)